src/controller.coffee

```diff
@@ -2866,6 +2866,17 @@ Editor::scrollCursorIntoPosition = ->
 
   @mainScroller.scrollLeft = 0
 
+# Moves the cursor to the end of the document and scrolls it into position
+# (in block and text mode)
+Editor::scrollCursorToEndOfDocument = ->
+  pos = @tree.end
+  while pos && !@validCursorPosition(pos)
+    pos = pos.prev
+  @setCursor(pos)
+  @scrollCursorIntoPosition()
+  @aceEditor.scrollToLine @aceEditor.session.getLength()
```

Comment:
Do we need to scroll both text and block mode, or can we detect which mode they're in and scroll only that mode?